zenoh-plugin-dds/src/dds_mgt.rs

```diff
@@ -119,6 +211,38 @@ impl Drop for DDSRawSample {
     }
 }
 
+impl fmt::Debug for DDSRawSample {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        #[cfg(feature = "dds_shm")]
+        {
+            if let Some(iox_chunk) = self.iox_chunk {
+                return write!(
+                    f,
+                    "[{:02x?}, {:02x?}]",
+                    self.data_as_slice(),
+                    iox_chunk.as_slice()
+                );
+            }
+        }
+        write!(f, "{:02x?}", self.data_as_slice())
+    }
+}
+
+impl From<DDSRawSample> for Value {
+    fn from(buf: DDSRawSample) -> Self {
+        #[cfg(feature = "dds_shm")]
+        {
+            if let Some(iox_chunk) = buf.iox_chunk {
+                let mut zbuf = ZBuf::default();
+                zbuf.push_zslice(ZSlice::from(buf.data_as_slice().to_vec()));
```

Comment:
Why to push a slice from data here? Isn't all in `iox_chunk`?
Please add a comment in code to explain, so we don't wonder in the future.

Reply:
No, when a shared memory buffer is received only the payload is in the `iox_chunk`. The header is still contained within the iovec data. This is why both are pushed into the buffer. I'll add a comment to clarify.